src/nodes.coffee

```diff
@@ -5342,6 +5369,17 @@ makeDelimitedLiteral = (body, options = {}) ->
     when other     then (if options.double then "\\#{other}" else other)
   "#{options.delimiter}#{body}#{options.delimiter}"
 
+sniffDirectives = (expressions, {notFinalExpression} = {}) ->
+  index = 0
+  lastIndex = expressions.length - 1
+  while index <= lastIndex and expression = expressions[index]
+    break if index is lastIndex and notFinalExpression
```

Comment:
We need a `notFInalExpression` option to make sure that in the case of a block that is only strings that look like directives, the final one is not considered a directive since it's implicitly returned

Eg:
```
->
  'not a directive'
```